### PR TITLE
Janus robustness

### PR DESCRIPTION
**Pull request by kolmanv**

### CODE REVIEW

File: backend/Gemfile.lock [hunk 1]
```diff
@@ -271,6 +272,7 @@ DEPENDENCIES
   faye-rails
   jbuilder (~> 2.0)
   jquery-rails
+  pg
```
Comment (kkavod) [inline backend/Gemfile.lock]: Remove this file from the repo

File: frontend/e2e/utils.js [hunk 2]
```diff
@@ -13,7 +13,7 @@ exports.login = function() {
   submit.click();
 
   var ret = false;
-  setTimeout(function() { ret = true; }, 2500);
+  setTimeout(function() { ret = true; }, 3500);
   browser.wait(function() { return ret; });
```
Comment (kkavod) [inline frontend/e2e/utils.js]: replace with browser.sleep()

File: frontend/gulp/common.js [hunk 3]
```diff
@@ -8,6 +8,13 @@
 
 var gutil = require('gulp-util');
 
+exports.argv = require('yargs').argv;
```
Comment (kkavod) [inline frontend/gulp/common.js]: Consider using process.argv http://stackoverflow.com/questions/4351521/how-do-i-pass-command-line-arguments-to-node-js

File: frontend/gulp/common.js [hunk 4]
```diff
@@ -8,6 +8,13 @@
...
+if ('port' in exports.argv) {
+  console.log('Using base port', exports.argv.port);
+} else {
+  throw "Port not defined, please specify --port=9876";
+}
```
Comment (kkavod) [inline frontend/gulp/common.js]: Consider moving from common.js to other place because this code is not common.

**Comment by kkavod** [comment]
LGTM, pls merge after considering all the comments

File: backend/Gemfile.lock (hunk 1 above)
Comment (kolmanv) [inline backend/Gemfile.lock]: Done.

File: frontend/gulp/common.js (hunk 3 above)
Comment (kolmanv) [inline frontend/gulp/common.js]: process.argv is an array and require('yargs').argv is parsed object.

**Comment by kolmanv** [comment]
All done.

**Pull request merged**